lib/AST/ASTContext.cpp

```diff
@@ -3100,7 +3107,7 @@ static void checkFunctionRecursiveProperties(Type Input,
   // constraint solving:
   //assert(!Input->getRecursiveProperties().isLValue()
   //       && "function should not take lvalues directly as parameters");
-  //assert(Result->getRecursiveProperties().isMaterializable()
+  //assert(Result->getRecursiveProperties().isLValue()
```

Comment:
This is false because opened types for mutable instance properties and subscripts have an lvalue result. You should just remove these lines